examples/src/main/java/alluxio/examples/UnderFileSystemCommonOperations.java

```diff
@@ -675,27 +673,24 @@ public void listStatusFileTest() throws IOException {
    * Test for listing large directory.
    */
   @RelatedS3Operations(operations = {"putObject", "listObjectsV2", "getObjectMetadata"})
-  public void listLargeDirectoryTest() throws IOException {
+  public void listLargeDirectoryTest() throws Exception {
     LargeDirectoryConfig config = prepareLargeDirectory();
     String[] children = config.getChildren();
 
     // Retry for some time to allow list operations eventual consistency for S3 and GCS.
     // See http://docs.aws.amazon.com/AmazonS3/latest/dev/Introduction.html and
     // https://cloud.google.com/storage/docs/consistency for more details.
-    // Note: not using CommonUtils.waitFor here because we intend to sleep with a longer interval.
-    UfsStatus[] results = new UfsStatus[] {};
-    for (int i = 0; i < 50; i++) {
-      results = mUfs.listStatus(config.getTopLevelDirectory());
-      if (children.length == results.length) {
-        break;
+    AtomicReference<UfsStatus[]> results = new AtomicReference<>();
```

Comment:
why do we need an atomic here?

Reply:
this was because we cannot set variables from within a closure, so I couldn't assign to `results`.